File: laravel-app/app/Models/UserAchievement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class UserAchievement extends Model
{
    /** @use HasFactory<\Database\Factories\UserAchievementFactory> */
    use HasFactory;
    protected $fillable = [
        'id',
        'user_id',
        'achievement_type_id',
        'earned_at',
        'progress_data'
    ];
    protected $casts = [
        'progress_data' => 'array'
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
    public function achievementType(): BelongsTo
    {
        return $this->belongsTo(AchievementType::class);
    }
}

File: netwinform-app/Models/UserAchievement.cs
using System;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace netwinform_app.Models
{
    public class UserAchievement
    {
        [JsonProperty("id")]
        public string Id { get; set; }
        [JsonProperty("user_id")]
        public string UserId { get; set; }
        [JsonProperty("achievement_type_id")]
        public string AchievementTypeId { get; set; }
        [JsonProperty("earned_at")]
        public string AchievementType { get; set; }
        [JsonProperty("progress_data")]
        public JObject ProgressData { get; set; }
    }
}
